omegatrade-app/frontend/src/app/shared/control-messages/control-messages.component.ts

```diff
@@ -0,0 +1,27 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { FormGroup, FormControl } from '@angular/forms';
+import { ValidationService } from './../validation.service';
+
+@Component({
+  selector: 'control-messages',
+  template: `
+  <div *ngIf="errorMessage !== null">{{errorMessage}}</div>
+  `
+})
+export class ControlMessagesComponent implements OnInit {
+
+  @Input() control: FormControl;
+  constructor() { }
+
+  ngOnInit(): void {
+  }
+  get errorMessage() {
+    for (let propertyName in this.control.errors) {
+      if (this.control.errors.hasOwnProperty(propertyName) && this.control.touched) {
```

Comment:
`for (let propertyName in this.control.errors) {`
Should have put check of undefined of parent property before direct accessing child property for the safer side, in-case of `this.control` is undefined or not an object it can give error

Reply:
This issue has been fixed.